src/algebra/group/GroupModule.lean

```diff
@@ -0,0 +1,87 @@
+import algebra.category.Group.basic
+import category_theory.endomorphism
+
+universe u₁
+
+open category_theory
+
+structure GroupModule (G : Group.{u₁}) :=
+(V : AddCommGroup.{u₁})
+(ρ : G ⟶ Group.of (Aut V))
+
+namespace GroupModule
+variable {G : Group.{u₁}}
+
+@[ext]
+structure hom (M N : GroupModule G) :=
+(hom : M.V ⟶ N.V)
+(comm' : ∀ g : G, (M.ρ g).hom ≫ hom = hom ≫ (N.ρ g).hom . obviously)
+
+restate_axiom hom.comm'
+
+namespace hom
+
+@[simps]
+def id (M : GroupModule G) : GroupModule.hom M M :=
+{ hom := 𝟙 M.V }
+
+@[simps]
+def comp {M N K : GroupModule G} (p : GroupModule.hom M N) (q : GroupModule.hom N K) :
+  GroupModule.hom M K :=
+{ hom := p.hom ≫ q.hom,
+  comm' := λ g, by rw [←category.assoc, p.comm, category.assoc, q.comm, ←category.assoc] }
+
+end hom
+
+instance : category (GroupModule G) :=
+{ hom := λ M N, hom M N,
+  id := λ M, hom.id M,
+  comp := λ M N K f g, hom.comp f g, }
+
+instance : concrete_category (GroupModule G) :=
+{ forget :=
+  { obj := λ M, M.V,
+    map := λ M N f, f.hom },
+  forget_faithful :=
+  { injectivity' := λ M N f g w,
+    begin
+      -- TODO make this less tedious
+      ext, apply add_monoid_hom.ext, intro x, change (M.V : Type u₁) at x,
+      dsimp at w,
+      rw w,
+    end } }
+
+instance has_forget_to_AddCommGroup : has_forget₂ (GroupModule G) AddCommGroup :=
+{ forget₂ :=
+  { obj := λ M, M.V,
+    map := λ M N f, f.hom } }
+
+@[simps]
+def map {G H : Group} (f : G ⟶ H) : GroupModule H ⥤ GroupModule G :=
```

Comment:
```suggestion
def comap {G H : Group} (f : G ⟶ H) : GroupModule H ⥤ GroupModule G :=
```

Reply:
Or otherwise maybe `res` for restriction... which is the math name.

Reply:
Changed to `res`.